Content.Shared/VendingMachines/SharedVendingMachineSystem.cs

```diff
@@ -65,27 +75,37 @@ public List<VendingMachineInventoryEntry> GetAvailableInventory(EntityUid uid, S
             return;
         }
 
-        var inventory = new Dictionary<string, VendingMachineInventoryEntry>();
-
-        foreach (var (id, amount) in entries)
-        {
-            if (_prototypeManager.HasIndex<EntityPrototype>(id))
-            {
-                inventory.Add(id, new VendingMachineInventoryEntry(type, id, amount));
-            }
-        }
-
+        Dictionary<string, VendingMachineInventoryEntry> inventory;
         switch (type)
         {
             case InventoryType.Regular:
-                component.Inventory = inventory;
+                inventory = component.Inventory;
                 break;
             case InventoryType.Emagged:
-                component.EmaggedInventory = inventory;
+                inventory = component.EmaggedInventory;
                 break;
             case InventoryType.Contraband:
-                component.ContrabandInventory = inventory;
+                inventory = component.ContrabandInventory;
                 break;
+            default:
+                return;
+        }
+
+        foreach (var (id, amount) in entries)
+        {
+            if (_prototypeManager.HasIndex<EntityPrototype>(id))
+            {
+                if (inventory.TryGetValue(id, out VendingMachineInventoryEntry? entry))
+                    // Prevent a machine's stock from going over three times
+                    // the prototype's normal amount. This is an arbitrary
+                    // number and meant to be a convenience for someone
+                    // restocking a machine who doesn't want to force vend out
+                    // all the items just to restock one empty slot without
+                    // losing the rest of the restock.
+                    entry.Amount = Math.Min(entry.Amount + amount, 3 * amount);
+                else
+                    inventory.Add(id, new VendingMachineInventoryEntry(type, id, amount));
+            }
```

Comment:
I would think the restock is just bringing it up to the normal amount.

Reply:
If the machine is not entirely empty, you lose the difference between the starting inventory and that amount. Is that not undesirable? It would be a waste of money.

Sometimes not all items are vended out when a restock is needed. It encourages players to vend out what items haven't been completely emptied, if those items are still desirable. This allows them to restock extra within a certain limit for the sake of convenience.